console/src/components/Services/RemoteSchema/Permissions/PermissionsTable.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { ChangeEvent } from 'react';
+import styles from '../../../Common/Permissions/PermissionStyles.scss';
+import PermTableHeader from '../../../Common/Permissions/TableHeader';
+import PermTableBody from '../../../Common/Permissions/TableBody';
+import { permissionsSymbols } from '../../../Common/Permissions/PermissionSymbols';
+import { findRemoteSchemaPermission } from './utils';
+import { RolePermissions, PermissionsTableProps, Permissions } from './types';
+
+const queryTypes = ['Permission'];
+
+const PermissionsTable: React.FC<PermissionsTableProps> = ({ ...props }) => {
+  const {
+    allRoles,
+    currentRemoteSchema,
+    permissionEdit,
+    isEditing,
+    bulkSelect,
+    readOnlyMode,
+    permSetRoleName,
+    permSetBulkSelect,
+    setSchemaDefinition,
+    permOpenEdit,
+    permCloseEdit,
+  } = props;
+
+  const allPermissions = currentRemoteSchema.permissions as Permissions[];
```

Comment:
Don't go around TypeScript compiler by doing type assertion. Make sure that `currentRemoteSchema.permissions` is properly typed instead.